assemble/bin/accumulo-service

```diff
@@ -230,14 +231,14 @@ function list_processes() {
     local pid_file
     local pid
     local port
-    pid_file="${ACCUMULO_PID_DIR}/accumulo-${process}.pid"
-    pid=$(<"$pid_file")
-    port=$(netstat -tnlp 2>/dev/null | grep "$pid" | awk '{print $4}' | awk -F ':' '{print $NF}' | paste -sd "," -)
-    # check that only a single port was seen
-    if (($(echo "$port" | grep -c -E '^[0-9]+(,[0-9]+)*$') != 1)); then
-      echo "ERROR unexpected ports $(hostname) process:$process pid:$pid ports:$port" >&2
-    else
+    pid_file="$ACCUMULO_PID_DIR/accumulo-$process.pid"
+    pid=$(<"$pid_file") # read the contents of the file into the $pid variable
+    port=$(ss -tnlp 2>/dev/null | grep -F "pid=$pid," | awk '{print $4}' | awk -F : '{print $NF}' | paste -sd,)
```

Comment:
The grep for `pid=$pid` is good, and I see why you added the `,` to the pattern, but the `fd` field is not guaranteed to always be there, and the pid isn't always guaranteed to be prior to it. Instead, you could grep for word boundaries, and then you don't need to look for the comma:

```suggestion
    port=$(ss -tnlp 2>/dev/null | grep -wF "pid=$pid" | awk '{print $4}' | awk -F : '{print $NF}' | paste -sd,)
```